### PR TITLE
Add some CLI tests on the CI

### LINKED ISSUE #42: Add basic CLI tests on the CI
The CI should have Parsec running with the Mbed Crypto provider and test various commands with the `parsec-tool` to check that it works properly, that the commands stay stable and that the output is correct.

### PR DESCRIPTION
**Pull request by hug-dev**
This commit makes Parsec run on the CI and execute some CLI commands.

Fix #42

### CODE REVIEW

File: tests/ci.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+./target/debug/parsec-tool list-authenticators
+./target/debug/parsec-tool list-providers
+./target/debug/parsec-tool list-opcodes -p 1
+./target/debug/parsec-tool create-ecc-key -k toto
+./target/debug/parsec-tool create-rsa-key -k tata
+./target/debug/parsec-tool list-keys
+./target/debug/parsec-tool export-public-key -k toto
+./target/debug/parsec-tool export-public-key -k tata
+./target/debug/parsec-tool generate-random -n 50
+./target/debug/parsec-tool sign -k toto "Les carottes sont cuites!"
```
Comment (ionut-arm) [inline tests/ci.sh:41]: Am I too much of a prick if I ask you to pipe the output of these to some `test_blabla.out` files and then do a diff between that and the "correct" output in `test_blabla.valid` which you commit here?

Comment (ionut-arm) [reply tests/ci.sh:41]: (I guess some of them can't be compared properly, e.g. the `generate-random`. Don't know if there's some way to at least check the format (and at least that it doesn't fail))

Comment (hug-dev) [reply tests/ci.sh:41]: YES

ok will do something like that, was trying to get away from it

Comment (hug-dev) [reply tests/ci.sh]: I added output checks for the commands that are somewhat stable. I guess ideally we should use `openssl` or other to check the decrypt/sign functions but we can maybe do that in subsequent PRs 😄 Also, we do test those methods in the Rust client and Parsec

Comment (ionut-arm) [reply tests/ci.sh]: > we can maybe do that in subsequent PRs

Yep, agree!

> we do test those methods in the Rust client and Parsec

Well, yes and no - we don't check the PEM or ASN.1 conversions we've added

**Review by ionut-arm — APPROVED** [review]

**Pull request merged**